kotlin-runtime/ftl-runtime/src/main/kotlin/xyz/block/ftl/schemaextractor/ExtractSchemaRule.kt

```diff
@@ -205,30 +205,32 @@ class SchemaExtractor(
   private fun extractIngress(): MetadataIngress? {
     return verb.annotationEntries.firstOrNull {
       bindingContext.get(BindingContext.ANNOTATION, it)?.fqName?.asString() == Ingress::class.qualifiedName
-    }?.let {
-      val sourcePos = it.getLineAndColumn()
-      val argumentLists = it.valueArguments.partition { arg ->
-        // Method arg is named "method" or is of type xyz.block.ftl.Method (in the case where args are
-        // positional rather than named).
-        arg.getArgumentName()?.asName?.asString() == "method"
-          || arg.getArgumentExpression()?.getType(bindingContext)?.fqNameOrNull()
-          ?.asString() == Method::class.qualifiedName
+    }?.let { annotationEntry ->
+      val sourcePos = annotationEntry.getLineAndColumn()
+
+      require(annotationEntry.valueArguments.size >= 3) {
+        "$sourcePos ${verb.name} @Ingress annotation requires at least 3 arguments"
+      }
+
+      val typeArg = requireNotNull(annotationEntry.valueArguments[0].getArgumentExpression()?.text) {
+        "$sourcePos Could not extract type from ${verb.name} @Ingress annotation"
       }
-      val methodArg = requireNotNull(argumentLists.first.single().getArgumentExpression()?.text) {
+      val methodArg = requireNotNull(annotationEntry.valueArguments[1].getArgumentExpression()?.text) {
         "$sourcePos Could not extract method from ${verb.name} @Ingress annotation"
       }
-      // NB: trim leading/trailing double quotes because KtStringTemplateExpression.text includes them
-      val pathArg = requireNotNull(argumentLists.second.single().getArgumentExpression()?.text?.trim { it == '\"' }) {
+      val pathArg = requireNotNull(annotationEntry.valueArguments[2].getArgumentExpression()?.text) {
```

Comment:
@worstell you might want to check my work here. I def just made some assumptions, so want to make sure it looks ok to you :)